### PR TITLE
Show meta index size in vast status

### PR DESCRIPTION
**Pull request by lava**
###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Review commits independently.

### CODE REVIEW

File: libvast/vast/address_synopsis.hpp [hunk 1]
```diff
@@ -90,6 +90,11 @@ class buffered_address_synopsis : public synopsis {
     ips_.insert(*addr_view);
   }
 
+  size_t size() const override {
+    return sizeof(buffered_address_synopsis)
+      + ips_.size()*sizeof(typename decltype(ips_)::node_type);
```
Comment (dominiklohmann) [inline libvast/vast/address_synopsis.hpp:95]: For span, we have a `span::size_bytes` function that explicitly returns the size in bytes. I think this and all the other new functions should be named `size_bytes` as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,10 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- 🎁 A new key 'meta-index-bytes' to the output appears in the output
+  of `vast status --detailed`.
+  [#1193](https://github.com/tenzir/vast/pull/1193)
```
Comment (dominiklohmann) [inline CHANGELOG.md:17]: Can you reword this so it's a proper sentence?

File: libvast/src/system/index.cpp [hunk 3]
```diff
@@ -239,6 +239,7 @@ index_state::status(status_verbosity v) const {
       // Hence the fallback to low-level primitives.
       layout_object.insert_or_assign(name, std::move(xs));
     }
+    put(stats_object, "meta-idx-bytes", meta_idx.size());
```
Comment (dominiklohmann) [inline libvast/src/system/index.cpp:242]: The changelog says `meta-index-bytes`, this says `meta-idx-bytes`. I prefer the former.

**Review by dominiklohmann — COMMENTED** [review]
Just an initial look at the code.

File: libvast/vast/meta_index.hpp [hunk 4]
```diff
@@ -46,17 +46,17 @@ pack(flatbuffers::FlatBufferBuilder& builder,
 } // namespace system
 
 /// Contains one synopsis per partition column.
-//  TODO: Turn this into a proper struct with its own `add()` function
-//        and its own `partition_synopsis.hpp` header.
-//        Then we could store this in the `active_partition_state` directly
-//        instead of using a meta_index with only one entry.
+//  TODO: Move this into a separate `paratition_synopsis.hpp` header.
```
Comment (dominiklohmann) [inline libvast/vast/meta_index.hpp:49]:
```suggestion
//  TODO: Move this into a separate `partition_synopsis.hpp` header.
```

File: libvast/vast/address_synopsis.hpp [hunk 5]
```diff
@@ -90,6 +90,11 @@ class buffered_address_synopsis : public synopsis {
     ips_.insert(*addr_view);
   }
 
+  size_t size_bytes() const override {
+    return sizeof(buffered_address_synopsis)
+      + ips_.size()*sizeof(typename decltype(ips_)::node_type);
```
Comment (dominiklohmann) [inline libvast/vast/address_synopsis.hpp:95]:
```suggestion
      + ips_.size() * sizeof(typename decltype(ips_)::node_type);
```

Two minor notes:
- The formatting (spaces around binary operators)
- Should this be `ips_.size_bytes()` instead`

**Review by dominiklohmann — APPROVED** [review]
LGTM. I ran this locally, and it works as expected.

I did not verify the calculated numbers, but the person I'd ask here would be @lava anyways.

File: libvast/vast/address_synopsis.hpp (hunk 5 above)
Comment (lava) [reply libvast/vast/address_synopsis.hpp:95]: Here `ips_` is just a plain `std::unordered_map`, so it doesn't have a `size_bytes()` function and we have to estimate size as number of nodes times the byte size per node.

Comment (dominiklohmann) [reply libvast/vast/address_synopsis.hpp:95]: Ah fair enough, should've checked that.

**Pull request merged**